src/tuesday/core/instrument_models/noise.py

```diff
@@ -0,0 +1,197 @@
+"""A module to add thermal noise to lightcones."""
+
+import warnings
+from collections.abc import Callable
+
+import astropy.units as un
+import numpy as np
+from astropy.constants import c
+from astropy.cosmology import Planck18
+from astropy.cosmology.units import littleh
+from py21cmsense import Observation
+from py21cmsense.conversions import dk_du, f2z
+from scipy.signal import windows
+
+
+def grid_baselines(uvws, freq, boxlength: un.Quantity, lc_shape, weights):
+    r"""Grid baselines in uv space.
+
+    Parameters
+    ----------
+    uvws : np.ndarray
+        Baselines in uv space with shape (N bls, N time offsets, 3).
+    freq : un.Quantity
+        Frequency at which the baselines are projected.
+    boxlength : un.Quantity
+        Length of the box in which the lightcone is defined.
+    lc_shape : tuple
+        Shape of the lightcone (Nx, Ny, Nz).
+        We assume that Nx = Ny.
+    weights : np.ndarray
+        Weights for each baseline group with shape (N bls).
+
+    Returns
+    -------
+    uvsum : np.ndarray
+        2D histogram of uv counts for one day
+        of observation with shape (Nu=Nx, Nv=Nx).
+
+    """
+    dx = float(boxlength.value) / float(lc_shape[0])
+    ugrid_edges = (
+        np.fft.fftshift(np.fft.fftfreq(lc_shape[0], d=dx))
+        * 2
+        * np.pi
+        / Planck18.h
+        * littleh
+        / un.Mpc
+    )  # h/Mpc
+
+    ugrid_edges = np.append(
+        ugrid_edges, ugrid_edges[-1:] + ugrid_edges[-1:] - ugrid_edges[-2], axis=0
+    )
+
+    ugrid_edges /= dk_du(f2z(freq))
+
+    weights = np.repeat(weights, uvws.shape[1])
+    uvws = uvws.reshape((uvws.shape[0] * uvws.shape[1], -1))
+    uvsum = np.histogram2d(uvws[:, 0], uvws[:, 1], bins=ugrid_edges, weights=weights)[0]
+
+    # add mirrored baselines to uv grid
+    uvsum += np.flip(uvsum)
+    # but they're not independent measurements
+    uvsum /= 2.0
+
+    return uvsum
+
+
+def thermal_noise(
+    observation: Observation,
+    freqs: np.ndarray,
+    boxlen: float,
+    lc_shape: tuple,
+    a_eff: un.Quantity | None = None,
+):
+    r"""
+    Calculate thermal noise RMS per integration.
+
+    Eqn 3 from Prelogovic+22 2107.00018 without the last sqrt term
+    That eqn comes from converting Eqn 9 in Ghara+16 1511.07448
+    that's a flux density [Jy] to temperature [mK],
+    but without the assumption of a circular symmetry of antenna distribution.
+
+    Parameters
+    ----------
+    observation : py21cmsense.Observation
+        Instance of `Observation`.
+    freqs : astropy.units.Quantity
+        Frequencies at which the noise is calculated.
+    boxlen : astropy.units.Quantity
+        Length of the box in which the lightcone is defined.
```

Comment:
Update this docstring to be the same as the one above, after you change it according to my suggestion.